core/src/autogluon/core/utils/early_stopping.py

```diff
@@ -0,0 +1,103 @@
+
+class AbstractES:
+    def update(self, cur_round, is_best=False) -> bool:
+        raise NotImplementedError
+
+    def early_stop(self, cur_round, is_best=False) -> bool:
+        raise NotImplementedError
+
+
+class SimpleES(AbstractES):
+    def __init__(self, stopping_rounds=10):
```

Comment:
Documentation would be helpful, and example of how user can specify this in predictor.fit().
Docstring may also want to explain stopping_rounds is equivalent to early-stopping "patience" as it's called in many libraries.

Reply:
The example of how this can be specified in predictor.fit() could additionally be converted into a unit-test which could also be good to have.

Reply:
Renamed stopping_rounds to patience. Added more documentation. Will leave the info on how to specify in .fit for a future PR, the goal is to have defaults strong enough that it should be very rare for a user to need to specify them manually.